build-tools/mono-runtimes/mono-runtimes.targets

```diff
@@ -72,6 +72,70 @@
         WorkingDirectory="$(MonoSourceFullPath)"
     />
   </Target>
+
+  <Target Name="_PrepareLlvmItems">
+    <ItemGroup>
+      <_LlvmSourceFile Include="$(LlvmSourceFullPath)\lib\**\*.cpp" />
+      <_LlvmHeaderFile Include="$(LlvmSourceFullPath)\lib\**\*.h" />
+      <_LlvmMakefileConfig Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Makefile.config')" />
+      <_LlvmConfigureStamp Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\.stamp-configure')" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <_LlvmArchive Include="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)\Release\lib\*.a" />
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' "/>
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ConfigureLlvm"
+          DependsOnTargets="_PrepareLlvmItems"
+          Inputs="$(LlvmSourceFullPath)\Makefile.config.in;$(LlvmSourceFullPath)\configure;@(_LlvmSourceFile);@(_LlvmHeaderFile)"
+          Outputs="@(_LlvmMakefileConfig);@(_LlvmConfigureStamp)"
+          Condition=" $(_LlvmNeeded) != '' ">
+    <MakeDir Directories="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)" />
+    <Exec
+        Command="%(_LlvmRuntime.ConfigureEnvironment) $(LlvmSourceFullPath)\configure %(_LlvmRuntime.ConfigureFlags)"
+        WorkingDirectory="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)" />
+    <Touch
+        Files="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)\Makefile.config"
+        />
+    <Touch
+        Files="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)\.stamp-configure"
+        AlwaysCreate="true"
+        />
+  </Target>
+  
+  <Target Name="_BuildLlvm"
+          DependsOnTargets="_ConfigureLlvm"
+          Inputs="@(_LlvmConfigureStamp);@(_LlvmSourceFile);@(_LlvmHeaderFile)"
+          Outputs="@(_LlvmArchive);@(_LlvmSourceBinary)"
+          Condition=" '$(_LlvmNeeded)' != '' ">
+    <Exec
+        Command="%(_LlvmRuntime.BuildEnvironment) make $(MakeConcurrency) all install"
```

Comment:
This is both awesome and horrifying, e.g. setting `PATH` before `make` so that it works.

That said, is there no way to instead `configure` LLVM so that it uses full paths to tools, thus removing the need to mangle `PATH` here?